### PR TITLE
Updated documentation for `amethyst_test`.

### PR DESCRIPTION
**Pull request by azriel91**
## Description

Documentation updates for `amethyst_test` (missed them in the migration PR, oops).

## PR Checklist

By placing an x in the boxes I certify that I have:

- [x] Ran `cargo test --all` locally if this modified any rs files.
- [x] Ran `cargo +stable fmt --all` locally if this modified any rs files.
- [x] Updated the content of the book if this PR would make the book outdated.
- **n/a** Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- **n/a** Added unit tests for new APIs if any were added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

### CODE REVIEW

File: amethyst_test/src/amethyst_application.rs [hunk 1]
```diff
@@ -131,14 +131,19 @@ where
     /// Returns the built Application.
     ///
     /// If you are intending to call `.run()` on the `Application` in a test, be aware that on
-    /// Linux, this will cause a segfault when `RenderBundle` is added and GL is using software
-    /// rendering, such as when using Xvfb or when the following environmental variable is set:
+    /// Linux, this will cause a segfault when `RenderingSystem` is added and GL software rendering
```
Comment (Frizi) [inline amethyst_test/src/amethyst_application.rs]: there is no GL in the rendering pipeline :P

Comment (azriel91) [reply amethyst_test/src/amethyst_application.rs]: ah, i know, but was lazy to remove the comment for the future-dev when `glow` is integrated (and sift through git history) 😛

Comment (azriel91) [reply amethyst_test/src/amethyst_application.rs]: fixed!

**Comment by torkleyy** [comment]
Thanks!
bors r+

**Pull request merged**